app/models/manager_refresh/inventory/persister.rb

```diff
@@ -14,11 +14,16 @@ def initialize(manager, target)
 
   # creates method on class that lazy initializes an InventoryCollection
   def self.has_inventory(options)
-    name = options[:association]
+    name = options[:association] || options[:model_class].name.pluralize.underscore
+
     define_method(name) do
       collections[name] ||= begin
         collection_options = options.dup
-        collection_options[:parent] ||= manager
+
+        unless collection_options[:strategy] == :local_db_find_references
```

Comment:
could we rather change this to 
```
collection_options[:parent] = manager unless collection_options.key(:parent)
```

the order of priorities to talk to the the db is :custom_db_finder, :parent & :association, :arel

so not providing a key of ^ makes it try the next